src/index.js

```diff
@@ -38,6 +47,9 @@ class PromisePool extends Emitter {
     }
 
     if (this._numActive >= this._numConcurrent) {
+      if (this._pending.length >= this._maxPending) {
+        throw new Error('cannot queue function in pool: the pool is likely being used in a manner that does not propagate backpressure');
```

Comment:
I would love if this error message explicitly said something like "to queue more than _n_ pending jobs, pass a higher value for the `maxPending` option" so that the dev's next step was obvious.

Reply:
Per 1-1, I'm worried that they'll do that without understanding its significance.

Ended up doing 0fd60e8 to make the next action clearer.